hail/src/main/scala/is/hail/expr/types/physical/PStruct.scala

```diff
@@ -199,90 +46,41 @@ final case class PStruct(fields: IndexedSeq[PField], override val required: Bool
     sb.append('}')
   }
 
-  override def _pretty(sb: StringBuilder, indent: Int, compact: Boolean) {
-    if (compact) {
-      sb.append("Struct{")
-      fields.foreachBetween(_.pretty(sb, indent, compact))(sb += ',')
-      sb += '}'
-    } else {
-      if (size == 0)
-        sb.append("Struct { }")
-      else {
-        sb.append("Struct {")
-        sb += '\n'
-        fields.foreachBetween(_.pretty(sb, indent + 4, compact))(sb.append(",\n"))
-        sb += '\n'
-        sb.append(" " * indent)
-        sb += '}'
-      }
-    }
-  }
+  def copy(fields: IndexedSeq[PField] = this.fields, required: Boolean = this.required): PStruct
 
-  def selectFields(names: Seq[String]): PStruct = {
-    PStruct(required,
-      names.map(f => f -> field(f).typ): _*)
-  }
+  override def fundamentalType: PStruct = ???
```

Comment:
shouldn't need = ???

Reply:
It's because of the fundamentalType definition on PType

Reply:
ah. Instead you can do something like:
```
def fundamentalType: PStruct = structFundamentalType
protected def structFundamentalType: PStruct
```

Reply:
ah nice